src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs

```diff
@@ -16,7 +15,7 @@ public static class ImageEstimatorsCatalog
         /// </summary>
         /// <param name="catalog">The transform's catalog.</param>
         /// <param name="columns">The name of the columns containing the image paths(first item of the tuple), and the name of the resulting output column (second item of the tuple).</param>
-        public static ImageGrayscalingEstimator Grayscale(this TransformsCatalog catalog, params (string input, string output)[] columns)
+        public static ImageGrayscalingEstimator ConvertToGrayscale(this TransformsCatalog catalog, params (string input, string output)[] columns)
```

Comment:
>ConvertToGrayscale [](start = 48, length = 18)

Why are we changing the name here? Are we not using the same name as the estimator?

Reply:
The names of the methods are usually PerformAction. ConvertToGrayscale" felt more informative than "Grayscale"

---
In reply to: [235503615](https://github.com/dotnet/machinelearning/pull/1682#discussion_r235503615) [](ancestors = 235503615)

Reply:
I mean why are we not using ImageGrayScalingEstimator?

---
In reply to: [235504239](https://github.com/dotnet/machinelearning/pull/1682#discussion_r235504239) [](ancestors = 235504239,235503615)

Reply:
There is a broader discussion for it here: [https://github.com/dotnet/machinelearning/issues/1690](https://github.com/dotnet/machinelearning/issues/1690) 

I do see your point. As @cesardelatorre pointed out, the most correct thing to do would be to call those CreateActionPerformingEstimator. 

We chose to go with just the ActionPerforming verb. 

 [https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods)

---
In reply to: [235504834](https://github.com/dotnet/machinelearning/pull/1682#discussion_r235504834) [](ancestors = 235504834,235504239,235503615)